sktime/transformations/series/detrend/tests/test_mstl.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+import pandas as pd
+from sktime.datasets import load_airline
+from sktime.transformations.series.detrend import MSTL
+
+def test_transform_returns_correct_components():
+    # TODO: See if we need to create a fresh time series dataset?
+    # np.random.seed(42)
+    # data = np.random.randn(100)
+    # index = pd.date_range(start="2020-01-01", periods=100, freq="D")
+    # series = pd.Series(data, index=index)
+
+    # Load our default test dataset
+    series = load_airline()
+    series.index = y.index.to_timestamp()
+    
+    # Initialize the MSTL transformer with specific parameters
+    transformer = MSTL(periods=[3,12], return_components=True)
+
+    # Fit the transformer to the data
+    transformer.fit(series)
+
+    # Transform the data
+    transformed = transformer.transform(series)
+
+    # Check if the transformed data has the expected components
+    assert 'transformed' in transformed.columns, "Transformed component missing"
+    assert 'trend' in transformed.columns, "Trend component missing"
+    assert 'resid' in transformed.columns, "Residual component missing"
+    # TODO: Specify "Seasonal_3" and "Seasonal_12" missing for our specific data or not?
+    assert 'seasonal_3' in transformed.columns, "Seasonal component missing" 
+    assert 'seasonal_12' in transformed.columns, "Seasonal component missing"
```

Comment:
You don't need to give a message like `"Seasonal component missing"` here or in above statements when writing in test cases. This message will not be used anywhere as said here (https://github.com/sktime/sktime/pull/6039#discussion_r1511496318). Non blocking and optional

Reply:
I actually find messages like these helpful.

More than once I've been digging through test failures, trying to understand what exactly is being tested there. This can be very helpful for other developers!

Reply:
noted!

Reply:
here's a recent example, where I turned `assert` into `ValueError`:
https://github.com/astrogilda/tsbootstrap/pull/83/files

Things also got tedious because `assert`-s of equalities do not print out the two sides of the equality, leading to additional debugging overhead.